GrandCentralBoard/Core/Widget.swift

```diff
@@ -6,16 +6,38 @@
 import UIKit
 
 
-public protocol Updateable : class {
+/// Class implementing this protocol will have a collection of sources implementing `UpdatingSource` protocol.
+public protocol HavingSources : class {
     var sources: [UpdatingSource] { get }
+}
+
+/// Ability to be updated by calling `update(source:)` method with a source implementing `UpdatingSource` protocol.
+public protocol Updateable : class {
     func update(source: UpdatingSource)
 }
 
-public protocol Widget: Updateable {
+/**
+  Class implementing this protocol have it's controlled `view` and has to implement `Updateable` and `HavingSources` protocols.
+ */
+public protocol WidgetControlling: Updateable, HavingSources {
```

Comment:
I have mixed feelings about change from `Widget` to `WidgetControlling`. I am not sure if everything needs to have a `able` or `ing` prefix. When I see WidgetControlling, then I think "ok, this class controls the widget", while it both holds a view and updates it. So I think it is a widget, not a controller of a widget. (But I am not sure about it all).